pkg/build/buildkit/buildx_test.go

```diff
@@ -144,3 +149,71 @@ func Test_flattenMap(t *testing.T) {
 		})
 	}
 }
+
+func Test_detectCustomBuildArgsUsed(t *testing.T) {
+	contents := `FROM scratch
+ARG BAR=1
+ARG CUSTOM_FOO=2
+ARG CUSTOM_FOO1_BAR=nope
+ARG CUSTOM_STUFF_THINGS
+ARG NOT_CUSTOM_ARG
+
+CMD ["echo", "stuff"]
+`
+	argNames, err := detectCustomBuildArgsUsed(contents)
+	require.NoError(t, err, "detectCustomBuildArgsUsed failed")
+
+	wantArgs := map[string]struct{}{
+		"CUSTOM_FOO":          {},
+		"CUSTOM_FOO1_BAR":     {},
+		"CUSTOM_STUFF_THINGS": {},
+	}
+	require.Equal(t, wantArgs, argNames)
+}
+
+// This value is exactly the max sized argument allowed
+//
+//go:embed testdata/max-arg.txt
+var maxArg string
+
+func TestBuilder_determineBuildArgs(t *testing.T) {
+	// This value goes over the limit of arg size
+	oversizedArg := maxArg + "oopstoobig"
+
+	ctx := context.Background()
+	c := config.NewTestConfig(t)
+	b := NewBuilder(c.Config)
+	m := &manifest.Manifest{
+		Custom: map[string]interface{}{
+			// these can cause a problem when used as a build arg in the Dockerfile
+			"BIG_LABEL":         oversizedArg,
+			"ANOTHER_BIG_LABEL": oversizedArg,
+		},
+	}
+	// First we use the standard template, that does not use any custom build arguments
+	c.TestContext.AddTestFileFromRoot("pkg/templates/templates/create/template.buildkit.Dockerfile", "/.cnab/Dockerfile")
+
+	// Try manually passing too big of a --build-arg, this should fail
+	opts := build.BuildImageOptions{
+		BuildArgs: []string{"BIG_BUILD_ARG=" + oversizedArg},
+	}
+	_, err := b.determineBuildArgs(ctx, m, opts)
+	assert.ErrorContains(t, err, "BIG_BUILD_ARG is longer than the max")
```

Comment:
is there a specific need here that we used `assert` when the rest is using `require`?

Reply:
https://github.com/getporter/porter/pull/2643

Reply:
Got it, make sense, thank you lots for the explanation, Carolyn!